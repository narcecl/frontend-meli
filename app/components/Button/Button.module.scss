.button{
    border-radius: 6px;
    display: inline-block;
    font-family: 'Proxima Nova', -apple-system, Roboto, Arial, sans-serif;
    font-size: 16px;
    font-weight: 400;
    height: 48px;
    line-height: 48px;
    padding: 0 24px;
    text-align: center;
    border: none;
    cursor: pointer;
    display: block;
    transition: all .3s ease;

    &--primary{
        background: #3483fa;
        color: #fff;

        &:hover{
            background: #2968c8;
        }
    }

    &--secondary{
        background: rgba(65,137,230,.15);
        color: #3483fa;

        &:hover{
            background: rgba(65,137,230,.2);
        }
    }

    &--full{
        width: 100%;
    }
}