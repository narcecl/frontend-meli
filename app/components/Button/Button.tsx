import styles from './Button.module.scss';
import classNames from 'classnames/bind';

export interface ButtonProps {
    label: string;
    type?: 'primary' | 'secondary';
    full?: boolean;
}

export const Button = (props: ButtonProps) => {
    const { label, type = 'primary', full = false } = props;
    const cx = classNames.bind(styles);

    const componentClasses = cx(
        'button',
        `button--${type}`, {
        'button--full': full,
    });
    
    return (
        <button type="button" className={componentClasses}>
            { label }
        </button>
    );
}

export default Button;