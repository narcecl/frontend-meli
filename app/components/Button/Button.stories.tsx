import type { Meta, StoryObj } from '@storybook/react';
import { Button } from './Button';

// More on how to set up stories at: https://storybook.js.org/docs/react/writing-stories/introduction
const meta: Meta<typeof Button> = {
    title: 'Core/Button',
    component: Button,
    tags: ['autodocs'],
    argTypes: {
        label: {
            description: 'Defines button text content.',
            control: 'text',
        },
        type: {
            description: 'Defines button style theme.',
            control: 'select',
        },
        full: {
            description: 'Defines if button get full width (of container).',
            control: 'boolean',
        }
    }
};

export default meta;
type Story = StoryObj<typeof Button>;

// More on writing stories with args: https://storybook.js.org/docs/react/writing-stories/args
export const Default: Story = {
    args: {
       label: 'Button',
       type: 'primary',
       full: false
    },
};