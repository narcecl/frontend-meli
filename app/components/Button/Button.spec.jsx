import { render, screen } from '@testing-library/react';
import '@testing-library/jest-dom';
import Button from './Button';

describe('Button', () => {
	it('Default render', () => {
		// Define the label for the button
		const label = 'Button';

		// Render the component
		const { asFragment  } = render(
			<Button label={label} />
		);
		
		// Check first render
		const firstRender = asFragment();
        expect(firstRender).toMatchSnapshot();

		// Check attributes, classnames and styles
		const buttonElement = screen.getByRole('button');
		expect(buttonElement).toBeInTheDocument();
		expect(buttonElement).toHaveTextContent(label);
		expect(buttonElement).toHaveClass('button');
		expect(buttonElement).toHaveClass('button--primary');
		expect(buttonElement).not.toHaveClass('button--secondary');
		expect(buttonElement).not.toHaveClass('button--full');
	});

	it('Render with custom props', () => {
		// Define the label for the button
		const label = 'Button';

		// Render the component
		const { asFragment } = render(
			<Button label={label} type="secondary" full />
		);

		// Check first render
		const firstRender = asFragment();
        expect(firstRender).toMatchSnapshot();

		// Check attributes, classnames and styles
		const buttonElement = screen.getByRole('button');
		expect(buttonElement).toBeInTheDocument();
		expect(buttonElement).toHaveTextContent(label);
		expect(buttonElement).toHaveClass('button');
		expect(buttonElement).toHaveClass('button--secondary');
		expect(buttonElement).toHaveClass('button--full');
		expect(buttonElement).not.toHaveClass('button--primary');
	});
});
