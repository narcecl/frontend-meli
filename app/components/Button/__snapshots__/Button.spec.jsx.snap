// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button Default render 1`] = `
<DocumentFragment>
  <button
    class="button button--primary"
    type="button"
  >
    Button
  </button>
</DocumentFragment>
`;

exports[`Button Render with custom props 1`] = `
<DocumentFragment>
  <button
    class="button button--secondary button--full"
    type="button"
  >
    Button
  </button>
</DocumentFragment>
`;
